sympy/functions/combinatorial/numbers.py

```diff
@@ -890,10 +890,10 @@ def _eval_expand_func(self, **hints):
                 p = p - u * q
                 if u.is_nonnegative and p.is_positive and q.is_positive and p < q:
                     k = Dummy("k")
-                    t1 = q * Sum(1 / (q * k + p), (k, 0, u))
-                    t2 = 2 * Sum(cos((2 * pi * p * k) / S(q)) *
-                                   log(sin((pi * k) / S(q))),
-                                   (k, 1, floor((q - 1) / S(2))))
+                    t1 = q * Sum(S.One / (q * k + p), (k, S.Zero, u))
```

Comment:
None of these should be necessary either. q, k, and p are already SymPy types. Even if they weren't, a better fix would be to make sure they are sympified wherever they are defined, rather than trying to chase every place where they are used and avoiding integer/integer divisions.

Reply:
One of the change was to remove the S from q. I think that adding S.One and S.Zero will speed things up, but at the cost of readability.

Reply:
In general, I think it would be nice to be able to bypass sympifying when one know that the object are already sympified (as in functions like this).